src/Google/Ads/GoogleAds/V0/Util/GoogleAdsErrors.php

```diff
@@ -0,0 +1,118 @@
+<?php
+/**
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Ads\GoogleAds\V0\Util;
+
+use \Google\Rpc\Status;
+use Google\Ads\GoogleAds\V0\Errors\GoogleAdsError;
+use Google\Ads\GoogleAds\V0\Errors\GoogleAdsFailure;
+use Google\Ads\GoogleAds\V0\Errors\ErrorLocation\FieldPathElement;
+
+final class GoogleAdsErrors
+{
+    /**
+     * Gets a list of all partial failure error messages for a given response. Operations are
+     * indexed from 0.
+     *
+     * <p>For example, given the following GoogleAdsFailure:
+     *
+     * <pre>
+     *   <code>
+     *     errors {
+     *       message: "Too low."
+     *       location {
+     *         field_path_elements {
+     *           field_name: "operations"
+     *           index {
+     *             value: 1
+     *           }
+     *         }
+     *         field_path_elements {
+     *           field_name: "create"
+     *         }
+     *         field_path_elements {
+     *           field_name: "campaign"
+     *         }
+     *       }
+     *     }
+     *     errors {
+     *       message: "Too low."
+     *       location {
+     *         field_path_elements {
+     *           field_name: "operations"
+     *           index {
+     *             value: 2
+     *           }
+     *         }
+     *         field_path_elements {
+     *           field_name: "create"
+     *         }
+     *         field_path_elements {
+     *           field_name: "campaign"
+     *         }
+     *       }
+     *     }
+     *   </code>
+     * </pre>
+     *
+     * A single GoogleAdsError instance would be returned for operation index 1 and 2, and an empty
+     * list otherwise.
+     *
+     * @param int operationIndex the index of the operation, starting from 0.
+     * @param Status partialFailureStatus a partialFailure status, with the detail list
+     *    containing GoogleAdsFailure instances.
+     * @return GoogleAdsError[] an array containing the
```

Comment:
Do you really need to indent this line?
Or at least, we should indent at later positions. This is a bit too early.